javascript/main.js

```diff
@@ -0,0 +1,119 @@
+const form = document.getElementById('sign-up-form');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const formData = new FormData(e.target);
+  const errors = validateForm(formData);
+
+  document.getElementById('email-error').innerText = errors.email;
+  document.getElementById('username-error').innerText = errors.username;
+  document.getElementById('password-error').innerText = errors.password;
+  document.getElementById('confirm-password-error').innerText =
+    errors.confirmPassword;
+
+  if (
+    errors.email ||
+    errors.username ||
+    errors.password ||
+    errors.confirmPassword
+  )
+    return;
+
+  // Loading state for 2 secs & success
+  form.innerHTML = `
+    <div>Loading...</div>
+  `;
+
+  setTimeout(() => {
+    form.innerHTML = `
+      <div>You're signed in ${formData.get('username')}!</div>
+    `;
+  }, 2000);
```

Comment:
Creo que usar el metodo innerHTML no es una buena practica, en este caso lo utilice por practicidad, ya que luego esto sera realizado con react conditional rendering.

Reply:
You are right.. in this case I would care that much..but you can try for example toggling code using dispaly:hidden or display:block too